client/version_control/plugins/publish/collect_version_control_login.py

```diff
@@ -20,24 +22,83 @@ class CollectVersionControlLogin(pyblish.api.ContextPlugin):
     order = pyblish.api.CollectorOrder + 0.4990
     targets = ["local"]
 
-
     def process(self, context):
         version_control = AddonsManager().get("version_control")
-        if not version_control or not version_control.enabled:
-            self.log.info("No version control enabled")
-            return
-
         project_name = context.data["projectName"]
-        project_setting = context.data["project_settings"]
-        conn_info = version_control.get_connection_info(project_name,
-                                                        project_setting)
+        project_settings = context.data["project_settings"]
+        if not self._is_addon_enabled(version_control, project_settings):
+            self.log.info(
+                "Version control addon is not enabled"
+                f" for project '{project_name}'"
+            )
+            return
 
+        conn_info = self._get_conn_info(
+            project_name, version_control, project_settings)
         context.data["version_control"] = conn_info
 
-        PerforceRestStub.login(conn_info["host"], conn_info["port"],
-                               conn_info["username"],
-                               conn_info["password"],
-                               conn_info["workspace_dir"])
+        PerforceRestStub.login(
+            conn_info["host"],
+            conn_info["port"],
+            conn_info["username"],
+            conn_info["password"],
+            conn_info["workspace_dir"]
+        )
 
         stream = PerforceRestStub.get_stream(conn_info["workspace_dir"])
         context.data["version_control"]["stream"] = stream
+        self.log.debug(f"stream::{stream}")
+
+    def _is_addon_enabled(self, version_control, project_settings):
```

Comment:
I would make this as function in addon utils so you can import it where needed.


```python
def is_version_control_enabled(project_settings):
        return project_settings.get("version_control", {}).get("enabled", False)
```

Reply:
Ok, extracted.